src/lib/logstore/log_store.cpp

```diff
@@ -402,10 +410,18 @@ void HomeLogStore::flush_sync(logstore_seq_num_t upto_seq_num) {
     HS_DBG_ASSERT_EQ(LogDev::can_flush_in_this_thread(), false,
                      "Logstore flush sync cannot be called on same thread which could do logdev flush");
 
+    std::unique_lock lk(m_single_sync_flush_mtx);
```

Comment:
I am not sure this is right as when we do 
```
m_sync_flush_cv.wait(lk, [this, upto_seq_num] { return !m_records.status(upto_seq_num).is_active; });
```
the lk will be unlock, it probabaly cannot prevent multiple `flush_sync` entries

Reply:
ok.... actually the lk passed to `m_sync_flush_cv` is `m_sync_flush_mtx`, then this is right